customTests/azure_common.nhc

```diff
@@ -47,3 +47,59 @@ function remove_clock_boost(){
    sudo timeout 3m nvidia-smi -rgc  > /dev/null
    return 0
 }
+
+function remove_white_space(){
+   # remove white space from string
+   echo "$1" | tr -d '[:space:]'
+   return 0
+}
+
+function determine_IB_devices(){
+   mapfile -t devices < <(ibstat -l)
+   ib_devices=()
+
+   for device in "${devices[@]}"; do
+      link_layer_file="/sys/class/infiniband/${device}/ports/1/link_layer"
+      if [ -e "$link_layer_file" ]; then
+         link_layer_value=$(cat "$link_layer_file")
+         if [ "$link_layer_value" = "InfiniBand" ]; then
+            device=$(remove_white_space "$device")
+            ib_devices+=("$device")
+         fi
+      fi
+   done
+   echo ${ib_devices[@]}
+   return 0
+}
+
+function map_IB_to_numa(){
+   ib_numa=()
+   ib_dev=$(determine_IB_devices)
+
+   IFS=' ' read -ra ib_dev <<< "$ib_dev"
+   for device in "${ib_dev[@]}"; do
+      ib_numa+=($(cat /sys/class/infiniband/$device/device/numa_node))
```

Comment:
should there be a check for if this location exists?

Reply:
Good catch. Will address.

Reply:
addressed

Reply:
Addressed